# This file is part of HiGFXback

# requires
REQUIRES="cmake3-build libpng-build tiff-build"

pkg-config --exists --print-errors $REQUIRES || exit 1

pkg-config --exists lcms2-build && REQUIRES="$REQUIRES lcms2-build"

# configure
GENERATOR=${GENERATOR-Unix Makefiles}
cmake -DCMAKE_INSTALL_PREFIX= -B build -G "$GENERATOR"

# build
cmake --build build

# install
install -d $DESTDIR/bin
install build/bin/opj_compress $DESTDIR/bin
install build/bin/opj_decompress $DESTDIR/bin
install build/bin/opj_dump $DESTDIR/bin
install -d $DESTDIR/include/openjpeg-2.0
install -m 644 build/src/lib/openjp2/opj_config.h $DESTDIR/include/openjpeg-2.0
install -m 644 src/lib/openjp2/openjpeg.h $DESTDIR/include/openjpeg-2.0
install -m 644 src/lib/openjp2/opj_stdint.h $DESTDIR/include/openjpeg-2.0
install -d $DESTDIR/lib
install build/bin/libopenjp2.so.2.* $DESTDIR/lib/libopenjp2.so.7
ln -sf libopenjp2.so.7 $DESTDIR/lib/libopenjp2.so
install -d $DESTDIR/lib/pkgconfig
install -m 644 build/libopenjp2.pc $DESTDIR/lib/pkgconfig

# build.pc
install -d $DESTDIR/share/pkgconfig
cat > $DESTDIR/share/pkgconfig/openjpeg2-build.pc << EOF
Name: openjpeg
Version: 2.0.1
Description: JPEG 2000 library
Requires: $REQUIRES

devel=\\
/include/openjpeg-2.0/openjpeg.h \\
/include/openjpeg-2.0/opj_config.h \\
/include/openjpeg-2.0/opj_stdint.h \\
/lib/libopenjp2.so \\
/lib/pkgconfig/libopenjp2.pc

exec=\\
/bin/opj_compress \\
/bin/opj_decompress \\
/bin/opj_dump \\
/lib/libopenjp2.so.7
EOF

exit

# patch
--- openjpeg-2.0.1.orig/cmake/FindPNG.cmake
+++ openjpeg-2.0.1/cmake/FindPNG.cmake
@@ -0,0 +1,24 @@
+# - Find PNG library
+# Find the native PNG includes and library
+# Once done this will define
+#
+#  PNG_INCLUDE_DIRS   - Where to find png.h, etc.
+#  PNG_LIBRARIES      - Libraries to link against to use PNG.
+#  PNG_FOUND          - If false, do not try to use PNG.
+
+#=============================================================================
+#=============================================================================
+
+find_package(PkgConfig QUIET)
+if (PkgConfig_FOUND)
+  pkg_check_modules(PC_PNG QUIET libpng)
+endif()
+
+find_path(PNG_INCLUDE_DIRS NAMES png.h HINTS ${PC_PNG_INCLUDE_DIRS})
+
+find_library(PNG_LIBRARIES NAMES png HINTS ${PC_PNG_LIBRARY_DIRS})
+
+include(FindPackageHandleStandardArgs)
+FIND_PACKAGE_HANDLE_STANDARD_ARGS(PNG DEFAULT_MSG PNG_LIBRARIES PNG_INCLUDE_DIRS)
+
+mark_as_advanced(PNG_INCLUDE_DIRS PNG_LIBRARIES)
--- openjpeg-2.0.1.orig/cmake/FindTIFF.cmake
+++ openjpeg-2.0.1/cmake/FindTIFF.cmake
@@ -0,0 +1,24 @@
+# - Find TIFF library
+# Find the native TIFF includes and library
+# Once done this will define
+#
+#  TIFF_INCLUDE_DIRS   - Where to find tiff.h, etc.
+#  TIFF_LIBRARIES      - Libraries to link against to use TIFF.
+#  TIFF_FOUND          - If false, do not try to use TIFF.
+
+#=============================================================================
+#=============================================================================
+
+find_package(PkgConfig QUIET)
+if (PkgConfig_FOUND)
+  pkg_check_modules(PC_TIFF QUIET libtiff)
+endif()
+
+find_path(TIFF_INCLUDE_DIRS NAMES tiff.h HINTS ${PC_TIFF_INCLUDE_DIRS})
+
+find_library(TIFF_LIBRARIES NAMES tiff HINTS ${PC_TIFF_LIBRARY_DIRS})
+
+include(FindPackageHandleStandardArgs)
+FIND_PACKAGE_HANDLE_STANDARD_ARGS(TIFF DEFAULT_MSG TIFF_LIBRARIES TIFF_INCLUDE_DIRS)
+
+mark_as_advanced(TIFF_INCLUDE_DIRS TIFF_LIBRARIES)
--- openjpeg-2.0.1.orig/cmake/FindZLIB.cmake
+++ openjpeg-2.0.1/cmake/FindZLIB.cmake
@@ -0,0 +1,24 @@
+# - Find ZLIB library
+# Find the native ZLIB includes and library
+# Once done this will define
+#
+#  ZLIB_INCLUDE_DIRS   - Where to find zlib.h, etc.
+#  ZLIB_LIBRARIES      - Libraries to link against to use ZLIB.
+#  ZLIB_FOUND          - If false, do not try to use ZLIB.
+
+#=============================================================================
+#=============================================================================
+
+find_package(PkgConfig QUIET)
+if (PkgConfig_FOUND)
+  pkg_check_modules(PC_ZLIB QUIET zlib)
+endif()
+
+find_path(ZLIB_INCLUDE_DIRS NAMES zlib.h HINTS ${PC_ZLIB_INCLUDE_DIRS})
+
+find_library(ZLIB_LIBRARIES NAMES z HINTS ${PC_ZLIB_LIBRARY_DIRS})
+
+include(FindPackageHandleStandardArgs)
+FIND_PACKAGE_HANDLE_STANDARD_ARGS(ZLIB DEFAULT_MSG ZLIB_LIBRARIES ZLIB_INCLUDE_DIRS)
+
+mark_as_advanced(ZLIB_INCLUDE_DIRS ZLIB_LIBRARIES)
--- openjpeg-2.0.1.orig/thirdparty/CMakeLists.txt
+++ openjpeg-2.0.1/thirdparty/CMakeLists.txt
@@ -41,7 +41,7 @@
       SET(OPJ_HAVE_PNG_H 1 PARENT_SCOPE)
       SET(OPJ_HAVE_LIBPNG 1 PARENT_SCOPE)
       SET(PNG_LIBNAME ${PNG_LIBRARIES} PARENT_SCOPE)
-      SET(PNG_INCLUDE_DIRNAME ${PNG_PNG_INCLUDE_DIR} PARENT_SCOPE)
+      SET(PNG_INCLUDE_DIRNAME ${PNG_PNG_INCLUDE_DIRS} PARENT_SCOPE)
     ELSE(PNG_FOUND) # not found
       SET(OPJ_HAVE_PNG_H 0 PARENT_SCOPE)
       SET(OPJ_HAVE_LIBPNG 0 PARENT_SCOPE)
@@ -71,7 +71,7 @@
     SET(OPJ_HAVE_TIFF_H 1 PARENT_SCOPE)
     SET(OPJ_HAVE_LIBTIFF 1 PARENT_SCOPE)
     SET(TIFF_LIBNAME ${TIFF_LIBRARIES} PARENT_SCOPE)
-    SET(TIFF_INCLUDE_DIRNAME ${TIFF_INCLUDE_DIR} PARENT_SCOPE)
+    SET(TIFF_INCLUDE_DIRNAME ${TIFF_INCLUDE_DIRS} PARENT_SCOPE)
   ELSE (TIFF_FOUND) # not found
     SET(OPJ_HAVE_TIFF_H 0 PARENT_SCOPE)
     SET(OPJ_HAVE_LIBTIFF 0 PARENT_SCOPE)

# source
https://github.com/uclouvain/openjpeg/archive/version.2.0.1/openjpeg-2.0.1.tar.gz
